### ISSUE TITLE
"bounds must be positive" Exception during ExtraTreesTrainer training in certain nodes

### ISSUE THREAD

**Issue by leonhardanderle** [oracle/tribuo#407]
**Describe the bug**
While training an ExtraTreesTrainer on a dataset with multiple features and high tree depth, the training process sometimes fails after successfully building hundreds of trees. The following exception is thrown:
```
bound must be positive
at org.tribuo.regression.rtree.impl.RegressorTrainingNode.buildRandomTree(RegressorTrainingNode.java:236)
at org.tribuo.regression.rtree.impl.RegressorTrainingNode.buildTree(RegressorTrainingNode.java:150)
```

**To Reproduce**
The following configuration was used:
```
var trainer = new ExtraTreesTrainer<>(
    new CARTRegressionTrainer(
        12,       // maxDepth: tree depth
        1.0f,     // minChildWeight
        0.0f,     // minImpurityDecrease
        0.5f,     // fractionFeaturesInSplit
        true,     // useRandomSplitPoints
        new MeanSquaredError(), // impurity
        42L       // seed
    ),
    new AveragingCombiner(),
    500,       // numTrees
    42L
);
```

**Additional context**

In RegressorTrainingNode.buildRandomTree:
```
// if there is only 1 inverted feature for this feature, it has only 1 value, so cannot be split
if (feature.size() == 1) {
    continue;
}

int splitIdx = rng.nextInt(feature.size() - 1);
```

**Observation / Hypothesis**
In certain nodes, there may be no values left for a selected feature.

In such cases, feature.size() == 0, which results in rng.nextInt(feature.size() - 1) → rng.nextInt(-1) → "bounds must be positive" exception.

The current check feature.size() == 1 only handles the case with exactly one value, not the empty case.

**Comment by leonhardanderle** [comment]
Thanks for the quick response!
I tried setting minImpurityDecrease to 1e-8 and also to a slightly larger value (1e-6), but unfortunately I still hit the same error with the same seed. Most of the time training works fine. I had to run about 10 different seeds with 500 trees each before the problem occurred again.
